overlord/snapstate/backend/copydata.go

```diff
@@ -59,7 +59,27 @@ func (b Backend) UndoCopySnapData(newInfo *snap.Info, oldInfo *snap.Info, meter
 		if err2 != nil {
 			logger.Noticef("Cannot remove common data directories for %q: %v", newInfo.Name(), err2)
 		}
+	} else {
+		err2 = b.untrashData(newInfo)
+		if err2 != nil {
+			logger.Noticef("Cannot restore data from trash for %q: %v", newInfo.Name(), err2)
```

Comment:
We should probably avoid leaking the "trash" term into the UI as people won't get it. How about:

`"Cannot restore original data for %q while undoing: %v"`